gerritlab/main.py

```diff
@@ -205,26 +207,70 @@ def create_merge_requests(repo, remote, local_branch):
                 title=title,
                 description=desp,
             )
+            proposed_new_mrs.append(mr)
             c.mr = mr
-            with timing("create_mrs"):
-                mr.create()
-                mr._sha = None
-            new_mrs.append(mr)
 
     # At this point we have one MR for each commit.
     # title/desc/target_branch may be out-of-date for preexisting MRs.
     # Augment the MR descriptions to include a list of related MRs.
     for c in commits_data:
         title, desc = generate_augmented_mr_description(commits_data, c)
         mr = c.mr
-
         mr.set_target_branch(c.target_branch)
         mr.set_title(title)
         mr.set_desc(desc)
-        with timing("update_mrs"):
-            if (mr.save() or mr._sha != c.commit.hexsha) and mr not in new_mrs:
-                updated_mrs.append(mr)
-                commits_to_pipeline_cancel.append(c.commit)
+        if mr not in proposed_new_mrs:
+            proposed_updated_mrs.append(mr)
+
+    if len(proposed_updated_mrs) == 0 and len(proposed_new_mrs) == 0:
+        print()
+        warn("No updated/new MRs.\n")
+        return
+    print_with_color("\nProposed MRs:\n", Bcolors.HEADER)
+    if len(proposed_updated_mrs) > 0:
+        print_with_color("MRs to update", Bcolors.OKGREEN)
+        for mr in proposed_updated_mrs:
+            mr.print_info(verbose=True)
+        print()
+    else:
```

Comment:
I think these `else` messages will be unhelpful most of the time.  I do like your idea of making a table of all involved MRs with a column for what will happen to it (unchanged/update/create).